modules/contact/src/actions/ContactAction.C

```diff
@@ -108,9 +108,22 @@ ContactAction::validParams()
       "normal_lm_scaling", 1., "Scaling factor to apply to the normal LM variable");
   params.addParam<Real>(
       "tangential_lm_scaling", 1., "Scaling factor to apply to the tangential LM variable");
-  params.addParam<bool>("use_dual", false, "Whether to use the dual mortar approach");
-
+  params.addParam<bool>(
+      "interpolate_normals",
+      true,
+      "Whether to interpolate the nodal normals (e.g. classic idea of evaluating field at "
+      "quadrature points). If this is set to false, then non-interpolated nodal normals will be "
+      "used, and then the _normals member should be indexed with _i instead of _qp");
```

Comment:
I think it's advisable to point out the `_i` vs. `_qp` part is aimed at code developers.

Reply:
yeah, this is something I would like to look into a bit more.